packages/zk-certificates/lib/merkleTree.ts

```diff
@@ -156,7 +157,29 @@ export class MerkleTree {
       leaf,
       pathElements: path,
       leafIndex,
-      root: this.root,
     };
   }
 }
+
+/**
+ * Calculates the root hash of a merkle tree from a proof.
+ * @param proof - Merkle proof to calculate the root hash from.
+ * @param poseidon - Poseidon instance to use for hashing.
+ * @returns Root hash of the merkle tree.
+ */
+export function getMerkleRootFromProof(
+  proof: MerkleProof,
+  poseidon: Poseidon,
+): string {
+  let currentNode = proof.leaf;
+  const dummyTree = new MerkleTree(0, poseidon);
+
+  // hash up the tree to the root
+  for (let i = 0; i < proof.pathElements.length; i++) {
+    const isNodeOnRight = (proof.leafIndex >> i) % 2 === 1;
+    const left = isNodeOnRight ? proof.pathElements[i] : currentNode;
```

Comment:
You could use syntax `const [left, right] = isNodeOnRight? [a, b] : [b, a]` if you think that it would be more readable. It definitely highlights that this is the same condition for both variables

Reply:
Good idea, thanks.